code/modules/mob/living/simple_animal/hostile/necro.dm

```diff
@@ -99,44 +128,80 @@
 
 	var/break_doors = CANT //If CAN, they can attempt to open doors. If CANPLUS, they break the door down entirely
 	var/health_cap = 250 //Maximum possible health it can have. Because screw having a 1000 health mob
-/*	wanted_objects = list(
-		/obj/machinery/light,        // Bust out lights
+	var/busy = FALSE //Stop spamming the damn doorsmash
+	wanted_objects = list(
+		/obj/machinery/light,
+		/obj/machinery/door        // Bust out lights
 	)
 	search_objects = 1
-
+/*
 /mob/living/simple_animal/hostile/necro/zombie/CanAttack(var/atom/the_target)
 	if(istype(the_target,/obj/machinery/light))
 		var/obj/machinery/light/L = the_target
 		// Not empty or broken
 		return L.status != 1 && L.status != 2
 	return ..(the_target)*/ //Too buggy, gets caught on terrain way too much
 
-/mob/living/simple_animal/hostile/necro/New(loc, mob/living/Owner, datum/mind/Controller)
-	..()
-	if(Owner)
-		faction = "\ref[Owner]"
-		friends.Add(Owner)
-		if(Controller)
-			mind = Controller
-			ckey = ckey(mind.key)
-			to_chat(src, "<big><span class='warning'>You have been risen from the dead by your new master, [Owner]. Do his bidding so long as he lives, for when he falls so do you.</span></big>")
-		var/ref = "\ref[Owner.mind]"
-		var/list/necromancers
-		if(!(Owner.mind in ticker.mode.necromancer))
-			ticker.mode:necromancer[ref] = list()
-		necromancers = ticker.mode:necromancer[ref]
-		necromancers.Add(Controller)
-		ticker.mode:necromancer[ref] = necromancers
-		ticker.mode.update_necro_icons_added(Owner.mind)
-		ticker.mode.update_necro_icons_added(Controller)
-		ticker.mode.update_all_necro_icons()
-		ticker.mode.risen.Add(Controller)
+/mob/living/simple_animal/hostile/necro/zombie/CanAttack(var/atom/the_target)
+	if(the_target == creator)
+		return 0
+	if(ismob(the_target))
+		var/mob/living/M = the_target
+		if(ishuman(the_target)) //Checking for food
+			var/mob/living/carbon/human/H = the_target
+			if(H.isDead())
+				if(check_edibility(H))
+					if(isjusthuman(H)) //Always chow down on humans
+						return the_target
+					else
+						if(H.health > -400)//So we're not chowing down on the same vox all day
+							return the_target
+						else
+							return 0
+				else
+					return 0
+			else
+				return ..(the_target)
+		else
+			if(M.isDead())
+				return 0
+
+	if(istype(the_target,/obj/machinery/door)) //Checking for doors
+		var/obj/machinery/door/D = the_target
+		if(can_open_door(D))
+			return the_target
+		else
+			return 0
+	if(istype(the_target,/obj/machinery/light))
+		var/obj/machinery/light/L = the_target
+		return L.status != 1 && L.status != 2
```

Comment:
Right, and how am I supposed to use defines that are in another file?

I've put the define name that's used in power/lighting.dm, and put them into necro.dm as 

return L.status != LIGHT_EMPTY && L.status != LIGHT_BROKEN

And it's giving me undefined var, assumedly because they're not in the same file.

What the fuck, PJB